typing/typetexp.ml

```diff
@@ -524,7 +529,10 @@ and transl_type_aux env policy mode styp =
                    List.map (transl_type env policy Alloc_mode.Global) stl)
             in
             List.iter (fun {ctyp_type; ctyp_loc} ->
-              match constrain_type_layout env ctyp_type Layout.value with
+              match
+                constrain_type_layout ~reason:Dummy_reason_result_ignored
```

Comment:
Actually, I've figured out what's potentially wrong with `Dummy_reason_result_ignored`: the layout can get preserved in the layout of a tyvar. So even if we check against `Ok _`, it might get printed someday. This is probably true everywhere we use `constraint_type_layout` (as opposed to `check_type_layout`).